symengine/sets.cpp

```diff
@@ -154,6 +156,30 @@ RCP<const Set> Interval::set_intersection(const RCP<const Set> &o) const
             return emptyset();
         }
     }
+    if (is_a<Integers>(*o)) {
+        if (is_a_Number(*start_) and is_a_Number(*end_)) {
+            auto first = SymEngine::ceiling(start_);
+            auto last = SymEngine::floor(end_);
+            if (eq(*first, *start_) and left_open_) {
+                first = add(first, integer(1));
+            }
+            if (eq(*last, *end_) and right_open_) {
+                last = add(last, integer(-1));
+            }
+            if (eq(*Lt(last, first), *boolTrue)) {
+                return emptyset();
+            }
+            set_basic container;
+            while (eq(*Ge(last, first), *boolTrue)) {
+                container.insert(first);
+                first = add(first, integer(1));
+            }
+            return finiteset(container);
+        } else {
+            return SymEngine::set_intersection(
```

Comment:
For an interval like,
`(x, y]` intersecting with Integers should give, `[floor(x) + 1, floor(y)]`
`(x, y)` intersecting with Integers should give, `[floor(x) + 1, ceiling(y)-1]`
`[x, y]` intersecting with Integers should give, `[celing(x), floor(y)]`
`[x, y)` intersecting with Integers should give, `[ceiling(x), ceiling(y)-1]`

Reply:
nvm, ignore the above